warp/jax_experimental/ffi.py

```diff
@@ -823,6 +905,292 @@ def ffi_callback(call_frame):
     jax.ffi.register_ffi_target(name, ffi_capsule, platform="CUDA")
 
 
+def jax_ad_kernel(
+    kernel,
+    num_outputs=1,
+    static_argnames=None,
+    vmap_method: Optional[str] = "broadcast_all",
+    launch_dim_arg_index: int = 0,
+    output_dims=None,
+):
+    """Create a JAX-callable from a Warp kernel with a custom VJP.
+
+    This helper wraps an existing Warp kernel in a JAX-compatible callable that
+    supports jit and grad via a custom VJP implemented with the FFI path.
+
+    Args:
+        kernel: The Warp kernel to wrap.
+        num_outputs: Number of output arrays produced by the kernel.
+        static_argnames: Optional iterable of argument names that should be treated
+            as static (non-differentiable) in JAX. If None, scalar (non-array) inputs
+            are treated as static by default.
+        vmap_method: How the callback transforms under jax.vmap.
+        launch_dim_arg_index: Index (in the kernel's argument list) of the input
```

Comment:
(Sorry was in wrong place before)
I don't this will work for everyone, for example if I want to do something for every pixel in an RGB image, then I want the kernel to be over the height,width but not channel dimension. Also, this is different than the existing jax_kernel interface. I wonder if it's better to force providing these explicitly?